WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -129,6 +130,7 @@ @implementation WPPostViewController
 - (void)dealloc
 {
     [_mediaGlobalProgress removeObserver:self forKeyPath:NSStringFromSelector(@selector(fractionCompleted))];
+    [self.post removeObserver:self forKeyPath:@"dateCreated"];
```

Comment:
I don't think this will work because by the time dealloc is called the post property is nil or it may not be the original post the observer was added to. I think the right way to handle this is to register/deregister an observer in a willset/didset equivalent of the post property.